src/Api/Vault/Controllers/CiphersController.cs

```diff
@@ -747,6 +747,33 @@ public async Task PostBulkCollections([FromBody] CipherBulkUpdateCollectionsRequ
         }
     }
 
+    [HttpPut("{id}/archive")]
+    [RequireFeature(FeatureFlagKeys.ArchiveVaultItems)]
+    public async Task PutArchive(Guid id)
+    {
+        var userId = _userService.GetProperUserId(User).Value;
+        var cipher = await GetByIdAsync(id, userId);
+        if (cipher == null)
+        {
+            throw new NotFoundException();
+        }
+
+        await _cipherService.ArchiveAsync(cipher);
+    }
+
+    [HttpPut("archive")]
+    [RequireFeature(FeatureFlagKeys.ArchiveVaultItems)]
+    public async Task PutArchiveMany([FromBody] CipherBulkArchiveRequestModel model)
+    {
+        if (!_globalSettings.SelfHosted && model.Ids.Count() > 500)
+        {
+            throw new BadRequestException("You can only archive up to 500 items at a time.");
+        }
+
+        var userId = _userService.GetProperUserId(User).Value;
```

Comment:
❓  I see you create a `HashSet<Guid>` below for the unarchive bulk endpoint. Should we do the same here? If not, we should probably make them consistent unless there's a reason they're different that I'm missing.

Also, if its it necessary, it might be beneficial to include a test case for it.

Reply:
Took another look, you're already creating a `HashSet` within the `CipherService` methods, so we don't need to do it again here.